grid_map_core/include/grid_map_core/iterators/CircleIterator.hpp

```diff
@@ -30,14 +30,14 @@ class CircleIterator
    * @param center the position of the circle center.
    * @param radius the radius of the circle.
    */
-  CircleIterator(const GridMap & gridMap, const Position & center, const double radius);
```

Comment:
I would like know what the motivation is for this change. It goes against const correctness. 
https://isocpp.org/wiki/faq/const-correctness

Can you share what compile warning this solves?

Reply:
Ah to be honest I didn't get these warnings myself. The title is a bit confusing, but this was just part of the effort to make rolling as up to date as possible as master.

Is it not advisable to fix these if we're not seeing them ourselves during `colcon build`? Then I would just port the major changes from [these](https://github.com/ANYbotics/grid_map/pull/486#issuecomment-2505618400), but it's a bit confusing to know what was a needed fix and what was just a code clean up.

Reply:
Yea, especially if the change breaks ABI, I don't see any strong reason to fold it in. Perhaps it's worth starting with any bugfixes that have associated tests?